lang/java/avro/src/main/java/org/apache/avro/util/Utf8.java

```diff
@@ -173,9 +178,15 @@ public int hashCode() {
     if (h == 0) {
       byte[] bytes = this.bytes;
       int length = this.length;
-      h = 1;
-      for (int i = 0; i < length; i++) {
-        h = h * 31 + bytes[i];
+      // If the array is filled, use the underlying JDK hash functionality.
+      // Starting with JDK 21, the underlying implementation is vectorized.
+      if (length > 7 && bytes.length == length) {
+        h = Arrays.hashCode(bytes);
```

Comment:
How does ` Arrays.hashCode(bytes)` behave when the length is smaller ?
Doesn't it fall back to serial execution internally for anything that is not vectorizable ?